packages/edit-site/src/components/dataviews/text-filter.js

```diff
@@ -14,6 +14,9 @@ export default function TextFilter( { filter, view, onChangeView } ) {
 	const [ search, setSearch, debouncedSearch ] = useDebouncedInput(
 		view.filters[ filter.id ]
 	);
+	useEffect( () => {
```

Comment:
Why this was needed?

Reply:
I'm not inside how text filters work, maybe @oandregal can clarify why this part is needed?

Reply:
This is how to reproduce the bug this effect fixed:

- having a page with title "My page title"
- filter the list of pages by the search component (e.g.: type "title")
- save the view
- reload the page 

At this point, we want the search component to show the searched string "title", otherwise the user will see a filtered list but would not know how (or whether) is filtered.

This happens because once `Search` is mounted, the `useDebouncedInput` hook is only updated upon changes on local state, ignoring new incoming values for `view.search`. I've noticed there are two network requests to the pages endpoint:

- the 1st with the default parameters (`view.search` is empty, which is the default state for `useDebouncedInput`), 
- the 2nd with the parameters from the persisted view (`view.search` is "title", the local states needs telling that ).

It seems this code is being refactored. It may be worth taking a look at this after to minimize the network requests, etc.